### PR TITLE
fix: prevent parent element onClick from triggering

### PR DESCRIPTION
**Pull request by brandedcow**
Resolves [brave-browser#36550](https://github.com/brave/brave-browser/issues/36550)

Summary
- Added preventDefault on button to prevent parent element onClick from triggering

Depends On
[Updated button props](https://github.com/brave/leo/pull/713)

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by fallaciousreasoning — COMMENTED** [review]
This looks good. Once https://github.com/brave/leo/pull/713 lands I'll update it in brave-core and we can kick of CI for this PR.

Thanks @brandedcow! (note: you'll need to sign this commit too)

**Comment by fallaciousreasoning** [comment]
Hey @brandedcow I don't think the commit before the merge is signed. I would do the following to resign and avoid needing a merge commit:

```
git reset --hard HEAD~1 # reset before the merge commit
git rebase brave/master # rebase onto master
```

because the rebase modifies the commit it should be signed. If not you can do a `git commit --amend --no-edit -S`

**Comment by brandedcow** [comment]
@fallaciousreasoning Let me know if the commit is properly signed, I reset my branch to head and made a signed commit.

**Comment by fallaciousreasoning** [comment]
Sweet, that looks good - unfortunately it'll be a while until we get this one merged 'cause I'll need to bump Nala first (and that's never very quick).

**Comment by fallaciousreasoning** [comment]
@brandedcow once https://github.com/brave/brave-core/pull/24379 lands you'll need to rebase (it'll probably take a few days at least) and I can kick off CI.

Sorry its such a process :laughing:

**Comment by fallaciousreasoning** [comment]
Mind rebasing?

**Comment by brandedcow** [comment]
@fallaciousreasoning Just updated the PR!

**Review by fallaciousreasoning — APPROVED** [review]

**Comment by brandedcow** [comment]
@fallaciousreasoning Hey, just checking in on this PR, I see one of the checks were unsuccessful and it is blocking the merge, do I need to do anything on my end?

**Comment by fallaciousreasoning** [comment]
Whoops forgot about this. I'll just see if I can work out why CI isn't passing

**Pull request merged**
